markdown/3.x/en/documentation/routing.md

```diff
@@ -35,6 +35,14 @@ function isSearchPage() {
 }
 ```
 
+#### NOTE
+
+Please do not use `search-immediate` on non-search pages or you will be immediately redirected to the search page.
+
+#### `before-search` event
+
+`before-search` event that is emitted by `ff-searchbox` can be also used to detect certain search events in order to perform redirects. More specifically, this event will occur for all searches that involve the searchbox, but will not occur for ASN, paging, sorting etc. requests. This might come in handy on category pages where entering a new query should trigger a new search that is not be restricted by the current category filter.
```

Comment:
Will there be an example of what _"detecting certain search events"_ might look like?

Reply:
I wasn't going to provide one.

Reply:
Please do. TLDR style is fine. Or a link to a resource to give users some inspiration.
I'm imagining users who are not that familiar with the library but would like to detect search events.

Reply:
That's a regular event that can be listened with `ffSearchBoxElem.addEventListener()`. I can't imagine user that want to use such advanced feature and can't use JS event listeners. But if you really think it's necessary I can add one.

Reply:
I'm not thinking about `ffSearchBoxElem.addEventListener("before-search", fn)`, if that is what you are referring to.
I'm more after the search events that can be detected by using `before-search`.
Currently I am stuck at

 _"When I use `before-search`"_,
- _"**how** do I detect certain search events"_
- _"**what** certain search events are available?"_

Reply:
By `certain search events` I mean all the search events that come from `ff-searchbox`
>`before-search` event [..] can be also used to detect **certain search events** in order to perform redirects. **More specifically**, this event will occur for **all searches that involve the searchbox**, but will not occur for ASN, paging, sorting etc. requests. So the fact that this event was fired makes the search event "certain". Do you think that's not clear enough?

Reply:
Now I see. Thanks for clarifying! I was genuinely at a loss.
With my current approach of thinking about this a slightly different order and the use of certain (^^) keywords would be clearer:
1) `before-search` is emitted by `ff-searchbox` whenever `ff-searchbox` issues request
2) in `before-search` listener you have the opportunity to respond to only certain/selected/required search requests (for redirects or whatev)
alternatively, _"you can choose if you respond to the event or not"_
3) asn etc are not related, don't trigger the event and so on

This could be followed by a super short example demonstrating the "event filtering". Mostly to give an introduction to what the event object looks like.
```js
ffSearchBoxElem.addEventListener("before-search", e => {
  if (e.whatever) {
    redirect();
  }
});
```